src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,164 @@
+% (C) Copyright 2020 CPP BIDS SPM-pipeline developers
+
+function convert3Dto4D
+  %
+  % Short description of what the function does goes here.
+  %
+  % USAGE::
+  %
+  %   [argout1, argout2] = templateFunction(argin1, [argin2 == default,] [argin3])
+  %
+  % :param argin1: (dimension) obligatory argument. Lorem ipsum dolor sit amet,
+  %                consectetur adipiscing elit. Ut congue nec est ac lacinia.
+  % :type argin1: type
+  % :param argin2: optional argument and its default value. And some of the
+  %               options can be shown in litteral like ``this`` or ``that``.
+  % :type argin2: string
+  % :param argin3: (dimension) optional argument
+  %
+  % :returns: - :argout1: (type) (dimension)
+  %           - :argout2: (type) (dimension)
+  %
+  % .. todo:
+  %
+  %    - expand to run through multiple subjs ans groups
+  %      (https://stackoverflow.com/questions/8748976/list-the-subfolders-in-a-folder-matlab-only-subfolders-not-files)
+  %    - generalize how to retrieve RT from sidecar json file
+  %    - saveMatlabBatch(matlabbatch, ['3Dto4D_dataType-' num2str(dataType) '_RT-' num2str(RT)], opt, subID);
+
+  % Set the folder where sequences folders exist
+  optSource.sourceDir = '/Users/barilari/Desktop/DICOM_UCL_leuven/renamed/sub-pilot001/ses-002/MRI';
```

Comment:
You might want to have a small sub-function that checks the content of `optSource` or set some defaults in it.

For example make sure that `optSource.sourceDir` is a valid directory or set an empty cell to `optSource.sequenceToIgnore`.